docker/kdc-registry.sh

```diff
@@ -0,0 +1,843 @@
+        --network ${network_name} \
+        --add-host=${zk_container_name}:${hwx_zk_ip} \
+        -d ${kafka_image}
+
+    checkStatus $? "Kafka"
+    echo "Apache Kafka started successfully. Time taken : ${SECONDS}s"
+}
+
+function startSchemaRegistry {
+    local sid="${1}"
+    local container_name="${2}"
+
+    isContainerExists ${container_name} "Schema Registry"
+    if [[ $? -eq 1 ]]; then
+        return 0;
+    fi
+
+    local db_type="${3}"
+    local db_name="schema_registry"
+    local user="registry_user"
+    local pwd="password"
+    local class_name=""
+    local url=""
+
+    case "${db_type}" in
+        mysql)
+            classname="com.mysql.jdbc.jdbc2.optional.MysqlDataSource"
+            url="jdbc:mysql://${mysql_container_name}/${db_name}"
+            ;;
+        oracle)
+            classname="oracle.jdbc.pool.OracleDataSource"
+            url="jdbc:oracle:thin:@${oracle_container_name}:1521:xe"
+            ;;
+        postgresql)
+            classname="org.postgresql.ds.PGSimpleDataSource"
+            url="jdbc:postgresql://${postgres_container_name}/${db_name}"
+            ;;
+        *)
+            echo "Invalid db type : ${db_type} not supported"
+            exit 1
+    esac
+
+    SECONDS=0
+    echo "Starting Schema Registry ${sid}"
+    docker run --name ${container_name} \
+        -h ${container_name} \
+        -e DB_TYPE=${db_type} \
+        -e DATA_SRC_CLASS_NAME=${classname} \
+        -e DB_URL=${url} \
+        -e DB_USER=${user} \
+        -e DB_PASSWORD=${pwd} \
+        -p 9010-9020:9090 \
+        -p 9030-9040:9091 \
+        --network ${network_name} \
+        -v ${sasl_secrets_dir}:/etc/registry/secrets \
+        -e REGISTRY_HEAP_OPTS="-Xmx1G -Xms1G -Djava.security.krb5.conf=/etc/registry/secrets/krb5.conf -Dsun.security.krb5.debug=true" \
+        -d ${registry_image}
+
+    checkStatus $? "Schema Registry"
+    echo "Schema Registry started successfully. Time taken : ${SECONDS}s"
+
+    echo "Connect with the below ports to start the registry client"
+    docker port ${container_name}
+}
+
+ask_yes_no() {
+    local prompt="${1}"
+    while true; do
+        read -r -p "${prompt} " response
+        case "${response}" in
+            [yY]|[yY][eE][sS]) _return=1; return;;
+            [nN]|[nN][oO]) _return=0; return;;
+            *);;
+        esac
+        echo "Please respond 'yes' or 'no'."
+        echo
+    done
+}
+
+ask_db_type() {
+    read -p "Which underlying db type to use ?
+            1. mysql
+            2. oracle
+            3. postgres
+        > " answer
+
+        case "${answer}" in
+            m|mysql) echo "mysql" ;;
+            p|postgres|postgresql) echo "postgresql";;
+            o|oracle) echo "oracle" ;;
+            *) echo "Invalid db type : ${answer}"; exit 1;;
+        esac
```

Comment:
When the user is prompted here, he might end up giving 1,2 or 3 instead of the name of the database. Can you remove the numbers before the Db type or take a corresponding number against the database type as the answer ?